main/character.py

```diff
@@ -9,15 +9,23 @@ def __init__(self, x, y, name, strength, defense, health_points):
         self.health_points = health_points
         self.health_points_max = health_points
         self.alive = True
+        self.stamina = 100
+        self.lowest_stamina_value = 10
+        self.regen_stamina_value = 15
         # self.image = pygame.image.load()
         # self.rect = self.image.get_rect()
         # self.rect.center = (x, y)
 
     def attack(self, target: 'Character'):
-        damage = max(0, self.strength - target.defense)
-
-        target.get_damage(damage)
-        print(f"{self.name} attacks {target.name} for {damage} damage!")
+        if self.stamina_check() == True:
```

Comment:
In this case `self.stamina_check()` returns `bool` value, so there is no need to compare this
```suggestion
        if self.stamina_check():
```